lib/Mojo/Pg/Database.pm

```diff
@@ -231,6 +231,20 @@ L<Mojo::Pg::Transaction/"commit"> has been called before it is destroyed.
   };
   say $@ if $@;
 
+  # Upsert rows in a transaction
+  eval {
+    my $tx = $db->begin;
+    $db->query('insert into wines values (?, ?)', 10, 'Chateau');
+    $tx->savepoint('sp1');
+    eval { $db->query('insert into wines values (?, ?)', 24, 'Chateau') };
+    if ($@) {
+      $tx->rollback_to('sp1');
+      $db->query('update wines set stock = stock + ? where winename = ?',
+        24, 'Chateau');
+    }
+    $tx->commit;
+  };
```

Comment:
I mostly use the savepoint for constraint violation. "on duplicate key ignore" in 9.5 will solve this problems. But this is not the only one use case savepoint.